### ISSUE TITLE
Weird target/callback pair passed to CCHttpRequest

### ISSUE THREAD

**Issue by j-jorge** [ezibyte/EziSocial-Plugin#5]
Hi,

In `EziSocialObject::downloadPhoto()`, you have the following code

```
CCHttpRequest* request = new CCHttpRequest();
request->setResponseCallback(parentNode, httpresponse_selector(EziSocialObject::onHttpRequestCompleted));
```

Where `parentNode` is a `CCLayer`. This code tells `CCHttpRequest` to call the `onHttpRequestCompleted()` method of `EziSocialObject` on an object of type `CCLayer`. This is nonsense.

Please fix this by passing making `EziSocialObject` to extend `CCObject` and pass `this`as the target of `setResponseCallback()`. The argument `parentNode` of the download functions will then be useless and can be removed.

**Comment by ezibyte** [comment]
I would appreciate if you can use words carefully: 'This is nonsense.' Thank you.

Let's focus on issue you have raised. 

Why EziSocialObject is not inheriting properties of CCObject?
There is only single reason to this - EziSocialObject is a singleton class that needs to maintains its reference even if all the scenes are destroyed. If we make it child of CCObject then either we need to make it children of any CCScene or CCLayer OR otherwise we need to enforce it to keep it in memory. As soon as scene changes, all the existing CCObjects destroyed itself. So if EziSocialObject is also a member of CCObject, it will also get destroyed.

So that way we need to reinitialise EziSocialObject again and again with every Scene change. Sure we can do that. But it has cost associated with it ->
1. Whatever the call made by EziSocialObject to the network will not be reachable back to EziSocialObject if the reference is destroyed on the scene changed. 
2. We need to consume extra CPU cycles with every time re-intialization process takes place.

**\* EziSocialObject uses only CCHTTPRequest for downloading the photos. Rest all network calls handle in Objective-C (for iOS) and in Java (for Android). 

Please share your thoughts.

**Comment by j-jorge** [comment]
Hi,

So, I'm OK with the idea of not making `EziSocialObject` to inherit from `CCObject`. Still, I think that requiring the caller of `getProfilePicForID()` to pass a `CCLayer` makes your API unclear, simply because to the question "Why should I pass a layer to this function?" the code answers by "because I need to call on it a function it does not have."

If you really need a `CCObject` here to handle the callback, I think you should create it yourself without exposing it to the user. Even better, you could implement your own subclass of `CCLayer` with the `onHttpRequestCompleted` function which would dispatch the result to `EziSocialObject`:

```
class DownloadHandler:
  public cocos2d::CCLayer
{
public:
  void onHttpRequestCompleted( cocos2d::CCNode *sender, void *data )
  {
    EziSocialObject::sharedObject()->onDownloadComplete( data );
  }
};
```

Then you could use an instance of this class for the call to `CCHttpRequest::setResponseCallback()`. Then, the API would be easier to use and the code would be clearer.

Regards,

**Comment by ezibyte** [comment]
Fixed in EziSocial v2.3
